modules/compiler/multi_llvm/include/multi_llvm/loop_utils.h

```diff
@@ -0,0 +1,44 @@
+// Copyright (C) Codeplay Software Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License") with LLVM
+// Exceptions; you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://github.com/codeplaysoftware/oneapi-construction-kit/blob/main/LICENSE.txt
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#ifndef MULTI_LLVM_LOOP_UTILS_H_INCLUDED
+#define MULTI_LLVM_LOOP_UTILS_H_INCLUDED
+
+#include <multi_llvm/llvm_version.h>
+
+#include "llvm/Transforms/Utils/LoopUtils.h"
+
+namespace multi_llvm {
+#if LLVM_VERSION_MAJOR >= 18
+
+inline llvm::Value *createSimpleTargetReduction(
```

Comment:
In the past for `multi_llvm` we've usually had one unconditional function with the *body* doing the `#if/#else`.

Something like:

``` cpp
inline llvm::Value *createSimpleTargetReduction(
    llvm::IRBuilderBase &B, const llvm::TargetTransformInfo *TTI,
    llvm::Value *Src, llvm::RecurKind RdxKind) {
#if LLVM_VERSION_MAJOR >= 18
  (void)TTI;
  return llvm::createSimpleTargetReduction(B, Src, RdxKind);
#else
  return llvm::createSimpleTargetReduction(B, TTI, Src, RdxKind);
#endif
}